test/infer/test_autoguide.py

```diff
@@ -670,3 +675,146 @@ def model():
     guide = AutoDiagonalNormal(model, init_loc_fn=init_to_value(values={"x": 1.0}))
     svi = SVI(model, guide, optim.Adam(1.0), Trace_ELBO())
     svi.init(random.PRNGKey(0))
+
+
+def get_optim():
+    scheduler = piecewise_constant_schedule(1.0e-3, {30 * 1000: 1.0e-4, 60 * 1000: 1.0e-5})
+    optimizer = optax.chain(
+        optax.scale_by_adam(),
+        optax.scale_by_schedule(scheduler),
+        optax.scale(-1.0)
+    )
+
+    return optimizer
+
+
+def test_autosemidais(N=32, P=4, sigma_obs=0.1, num_steps=90 * 1000):
+    X = RandomState(0).randn(N, P)
+    Y = X[:, 0] - 0.5 * X[:, 1] + sigma_obs * RandomState(1).randn(N)
+
+    def global_model():
+        return numpyro.sample("theta", dist.Normal(jnp.zeros(P), 1).to_event(1))
+
+    def local_model(subsample_size, theta):
+        with numpyro.plate("N", N, subsample_size=subsample_size):
+            X_batch = numpyro.subsample(X, event_dim=1)
+            Y_batch = numpyro.subsample(Y, event_dim=0)
+            tau = numpyro.sample("tau", dist.Gamma(5.0, 5.0))
+            numpyro.sample("obs", dist.Normal(X_batch @ theta, sigma_obs / jnp.sqrt(tau)),
+                           obs=Y_batch)
+
+    model8 = lambda: local_model(8, global_model())
+    model16 = lambda: local_model(16, global_model())
+
+    base_guide = AutoNormal(global_model)
+    guide16 = AutoSemiDAIS(model16, partial(local_model, 16), base_guide, K=4, eta_max=0.25, eta_init=0.005)
+    svi_result16 = SVI(model16, guide16, get_optim(), Trace_ELBO()).run(random.PRNGKey(0), num_steps)
+    samples16 = guide16.sample_posterior(random.PRNGKey(1), svi_result16.params)
+    assert samples16["theta"].shape == (P,) and samples16["tau"].shape == (16,)
+    dais_elbo16 = jax.vmap(lambda k: -Trace_ELBO().loss(k, svi_result16.params, model16, guide16))(random.split(random.PRNGKey(0), 5000)).mean().item()
+
+    print("svi_result16.params[auto_eta_coeff].mean()", svi_result16.params['auto_eta_coeff'].mean(), " (should be positive)")
+    print("svi_result16.params[auto_eta].mean()", svi_result16.params['auto_eta0'].mean())
+
+    guide8 = AutoSemiDAIS(model8, partial(local_model, 8), base_guide, K=4, eta_max=0.25, eta_init=0.005)
+    with handlers.seed(rng_seed=0):
+        guide8()  # initialize guide since we are not training this guide
+        # instead we reuse svi_result16 parameters
+    samples8 = guide8.sample_posterior(random.PRNGKey(1), svi_result16.params)
+    assert samples8["theta"].shape == (P,) and samples8["tau"].shape == (8,)
+    dais_elbo8 = jax.vmap(lambda k: -Trace_ELBO().loss(k, svi_result16.params, model8, guide8))(random.split(random.PRNGKey(0), 5000)).mean().item()
```

Comment:
is there any reason reusing `svi_result16` like this would be incorrect?

Reply:
I'm not sure. It seems to be valid.

Reply:
is it possible the parameters get mixed up somehow? so that the local parameters get assigned to the wrong datapoints? e.g. `z_0_loc`